src/js/components/tweetBlock.js

```diff
@@ -1,23 +1,25 @@
 import React from 'react';
 
 export default class TweetBlock extends React.Component {
+    constructor(props){
+        super(props);
+    }
+    
     render(){
-        let tweetLink = "https://twitter.com/" + 
-                        this.props.tweet.user.twitterHandle+ "/status/" + 
-                        this.props.tweet.id;
-            let readableDate = new Date(this.props.tweet.createdAt);
-            let dateArr = readableDate.toString().split(" ");
-            let dateOut = dateArr[1] + " " + dateArr[2];
+        let tweetLink = `https://twitter.com/${this.props.tweet.user.twitterHandle}/status/${this.props.tweet.id}`;
+        let readableDate = new Date(this.props.tweet.createdAt);
+        let dateArr = readableDate.toString().split(" ");
+        let dateOut = dateArr[1] + " " + dateArr[2];
             return (
             <div className="tweet-block">
                 <div className="user-div">
-                    <img src={this.props.tweet.user.profileImageUrl} id="user-icon"></img>
+                    <img src={this.props.tweet.user.profileImageUrl} className="user-icon"></img>
                     <div className="user-name">
                         {this.props.tweet.user.name}
                     </div>
-                    <div className="user-handle">
+                    <div className="user-handle" hidden={this.props?.hidden?.handle}>
```

Comment:
why the decision to just hide it?

Reply:
React's conditional rendering makes the component less readable in my opinion (and also doesn't seem to work the way it is documented). I will give it another shot, but this seemed like a pretty good solution overall.